crypto/src/key/secp256k1/mod.rs

```diff
@@ -0,0 +1,353 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::hash::{Blake2b32Stream, StreamHasher};
+use crate::random::{CryptoRng, Rng};
+use bitcoin::secp256k1;
+use serialization::{Decode, Encode};
+use zeroize::Zeroize;
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+pub enum Secp256k1KeyError {
+    InvalidData,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+// TODO implement zeroize
+pub struct Secp256k1PrivateKey {
+    key_data: secp256k1::SecretKey,
+}
+
+impl Encode for Secp256k1PrivateKey {
+    fn size_hint(&self) -> usize {
+        self.key_data.as_ref().size_hint()
+    }
+
+    fn encode_to<T: serialization::Output + ?Sized>(&self, dest: &mut T) {
+        self.key_data.as_ref().encode_to(dest);
+    }
+
+    fn encode(&self) -> Vec<u8> {
+        self.key_data.as_ref().encode()
+    }
+
+    fn using_encoded<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
+        self.key_data.as_ref().using_encoded(f)
+    }
+
+    fn encoded_size(&self) -> usize {
+        self.key_data.as_ref().encoded_size()
+    }
+}
+
+impl Decode for Secp256k1PrivateKey {
+    fn decode<I: serialization::Input>(input: &mut I) -> Result<Self, serialization::Error> {
+        let mut v = <[u8; secp256k1::constants::SECRET_KEY_SIZE]>::decode(input)?;
+        let result = secp256k1::SecretKey::from_slice(&v)
+            .map(|r| Secp256k1PrivateKey { key_data: r })
+            .map_err(|_| serialization::Error::from("Private Key deserialization failed"));
```

Comment:
Could we make the error more informative, something like the following:

```suggestion
            .map_err(|e| serialization::Error::from(format!("Private Key deserialization failed: {e}")));
```

(Not tested, please don't just blindly apply the change)